kong/plugins/http-log/handler.lua

```diff
@@ -18,68 +19,172 @@ HttpLogHandler.PRIORITY = 12
+-- @return error message if there was an error
+local function send_payload(self, conf, payload)
+  local method = conf.method
+  local timeout = conf.timeout
+  local keepalive = conf.keepalive
+  local content_type = conf.content_type
+  local http_endpoint = conf.http_endpoint
+
+  local ok, err
+  local parsed_url = parse_url(http_endpoint)
+  local host = parsed_url.host
+  local port = tonumber(parsed_url.port)
+
+  local httpc = http.new()
+  httpc:set_timeout(timeout)
+  ok, err = httpc:connect(host, port)
+  if not ok then
+    kong.log.err("failed to connect to ", host, ":", tostring(port), ": ", err)
+    return false
+  end
+
+  if parsed_url.scheme == "https" then
+    local _, err = httpc:ssl_handshake(true, host, false)
+    if err then
+      kong.log.err("failed to do SSL handshake with ",
+                   host, ":", tostring(port), ": ", err)
+      return false
+    end
+  end
+
+  local res, err = httpc:request({
+    method = method,
+    path = parsed_url.path,
+    query = parsed_url.query,
+    headers = {
+      ["Host"] = parsed_url.host,
+      ["Content-Type"] = content_type,
+      ["Content-Length"] = #payload,
+      ["Authorization"] = parsed_url.userinfo and (
+        "Basic " .. ngx_encode_base64(parsed_url.userinfo)
+      ),
+    },
+    body = payload,
+  })
+  if not res then
+    kong.log.err("failed request to ", host, ":", tostring(port), ": ", err)
+    return false
+  end
+
+  -- always read response body, even if we discard it without using it on success
+  local response_body = res:read_body()
+  local success = res.status < 400
+
+  if not success then
+    kong.log.err("request to ", host, ":", tostring(port),
+                 " returned status code ", tostring(res.status), " and body ",
+                 response_body)
+  end
+
+  ok, err = httpc:set_keepalive(keepalive)
+  if not ok then
+    kong.log.err("failed keepalive for ", host, ":", tostring(port), ": ", err)
+  end
+
+  return success
+end
+
+
+local function json_array_concat(entries)
+  return "[" .. table_concat(entries, ",") .. "]"
+end
+
+
+local function get_queue_id(conf)
+  return fmt("%s:%s:%s:%s:%s:%s",
+             conf.http_endpoint,
+             conf.method,
+             conf.content_type,
+             conf.timeout,
+             conf.keepalive,
+             conf.retry_count,
+             conf.queue_size,
+             conf.flush_timeout)
+end
+
+
+-- Only provide `name` when deriving from this class,
+-- not when initializing an instance.
+function HttpLogHandler:new(name)
+  name = name or "http-log"
+  HttpLogHandler.super.new(self, name)
+
+  self.name = name
 end
 
 
 function HttpLogHandler:log(conf)
   HttpLogHandler.super.log(self)
 
-  local route_id = conf.route_id or "global"
-  local buf = buffers[route_id]
-  if not buf then
-
-    if conf.queue_size == nil then
-      conf.queue_size = 1
+  local entry = cjson_encode(basic_serializer.serialize(ngx))
+
+  local queue_id = get_queue_id(conf)
```

Comment:
and change this to:

```lua
  local queue_id = queue_id_cache[conf]
```

Reply:
I'm going to opt out to this one for the same reason I mentioned before: I'm not 100% positive the tables are not duplicated.